app/controllers/submissions_controller.rb

```diff
@@ -24,40 +22,27 @@ def index
     end
     @submissions = @submissions.order(created_at: :desc).page(params[:page]).per_page(20)
     # TODO: fix submission permissions
-
-    respond_to do |format|
-      format.html # index.html.erb
-      format.xml { render xml: @submissions }
-    end
   end
 
-  # GET /submissions/1
-  # GET /submissions/1.xml
   def show
     @submission = Submission.find(params[:id])
     authorize @submission, :show?
-    respond_to do |format|
-      format.html # show.html.erb
-      format.xml { render xml: @submission }
+
+    if params.key?(:debug) && current_user.is_admin?
+      render xml: @submission
+    else
+      render :show
```

Comment:
If we're keeping this, I'd prefer keeping the `.xml` syntax if possible rather than changing to `?debug`, since it's clearer about the response format and makes it easier for other formats to be added in the future (e.g. JSON). Potentially either of these?

```suggestion
    respond_to do |format|
      format.html # show.html.erb
      format.xml { render xml: @submission } if current_user.is_admin?
    end
```

```suggestion
    if request.format.xml? && current_user.is_admin?
      render xml: @submission
    else
      render :show
```

Otherwise I don't mind if we just remove the XML response from here as well for now (I think the cases you found in the logs were probably just me testing that from when you originally asked about this).